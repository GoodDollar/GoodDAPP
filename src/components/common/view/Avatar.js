// @flow
import React, { useMemo } from 'react'
import { TouchableOpacity, View } from 'react-native'
import { Avatar } from 'react-native-paper'

import UnknownProfileSVG from '../../../assets/unknownProfile.svg'
import { withStyles } from '../../../lib/styles'
import useOnPress from '../../../lib/hooks/useOnPress'
import { isValidImage } from '../../../lib/utils/image'

/**
 * Touchable Avatar
 * @param {Props} props
 * @param {Function} [props.onPress]
 * @param {String} [props.source]
 * @param {Object} [props.style]
 * @param {Number} [props.size=34]
 * @returns {React.Node}
 */
const CustomAvatar = ({ styles, style, source, onPress, size, imageSize, children, unknownStyle, ...avatarProps }) => {
  const _onPress = useOnPress(onPress)

  const _source = useMemo(() => isValidImage(source), [source])

  return (
    <TouchableOpacity
      activeOpacity={1}
      disabled={!onPress}
      onPress={_onPress}
      style={[styles.avatarContainer, { width: size, height: size, borderRadius: size / 2 }, style]}
      underlayColor="#fff"
    >
      {_source ? (
        <Avatar.Image
          size={imageSize || size - 2}
          source={_source}
          style={{ backgroundColor: 'rgba(0, 0, 0, 0)' }}
          {...avatarProps}
        />
      ) : (
        <View
          style={[{ width: size, height: size, backgroundColor: 'rgba(0, 0, 0, 0)' }, unknownStyle]}
          {...avatarProps}
        >
          <UnknownProfileSVG />
        </View>
      )}
      {children}
    </TouchableOpacity>
  )
}

CustomAvatar.defaultProps = {
  size: 42,
}

const getStylesFromProps = ({ theme }) => ({
  avatarContainer: {
    backgroundColor: theme.colors.gray50Percent,
    borderWidth: 1,
    borderColor: theme.colors.gray80Percent,
    alignItems: 'center',
    justifyContent: 'center',
  },
})

export default withStyles(getStylesFromProps)(CustomAvatar)
