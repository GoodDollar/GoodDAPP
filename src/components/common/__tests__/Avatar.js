import React from 'react'
import renderer from 'react-test-renderer'
import Avatar from '../Avatar'

// Note: test renderer must be required after react-native.

describe('Avatar', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<Avatar />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<Avatar />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
