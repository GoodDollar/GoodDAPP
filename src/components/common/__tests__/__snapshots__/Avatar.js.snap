// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,1.00)",
        "borderBottomLeftRadius": "17px",
        "borderBottomRightRadius": "17px",
        "borderTopLeftRadius": "17px",
        "borderTopRightRadius": "17px",
        "height": "34px",
        "width": "34px",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitFlexBasis": "auto",
          "borderBottomLeftRadius": "17px",
          "borderBottomRightRadius": "17px",
          "borderTopLeftRadius": "17px",
          "borderTopRightRadius": "17px",
          "flexBasis": "auto",
          "height": "34px",
          "msFlexPreferredSize": "auto",
          "overflowX": "hidden",
          "overflowY": "hidden",
          "width": "34px",
          "zIndex": 0,
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.00)",
            "backgroundPosition": "center",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
            "bottom": "0px",
            "height": "100%",
            "left": "0px",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
            "width": "100%",
            "zIndex": -1,
          }
        }
      />
    </div>
  </div>
</div>
`;
